Ensure the ServiceAccount/Role/ClusterRole resources are created in order

Sgitario opened the pull request:
Fix https://github.com/quarkusio/quarkus/issues/32640 
Partially reverts https://github.com/quarkusio/quarkus/pull/32208

iocanel commented:
Given that we allow users to bring their own resources, it's possible to end up with dependency chains that are contains 3 or more items (e.g. Namespace -> ConfigMap -> Deployment).

I suggest we just order resources in a way similar to: https://github.com/superbrothers/ksort/blob/af6d8f61cdc596ef660c1ffd64c6ca8604b0cb91/kind_sorter.go#L29

Sgitario commented:
> Given that we allow users to bring their own resources, it's possible to end up with dependency chains that are contains 3 or more items (e.g. Namespace -> ConfigMap -> Deployment).
> 
> I suggest we just order resources in a way similar to: https://github.com/superbrothers/ksort/blob/af6d8f61cdc596ef660c1ffd64c6ca8604b0cb91/kind_sorter.go#L29

+100 
PR updated with this list.

maxandersen commented:
marking as backport as it result in app deployment error.

iocanel commented on extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/QuarkusFileWriter.java:
```diff
@@ -0,0 +1,103 @@
+            "RoleList",
+            ROLE_BINDING,
+            "RoleBindingList",
+            "Service",
```
```suggestion
            "Service",
            "ImageStream",
            "BuildConfig",
```

Sgitario replied:
Done

iocanel reviewed (changes requested):
Looks good, but we should also include `ImageStream` and `BuildConfig` resources.

Sgitario committed:
Ensure the ServiceAccount/Role/ClusterRole resourcs are created in order

Fix https://github.com/quarkusio/quarkus/issues/32640
Partially reverts https://github.com/quarkusio/quarkus/pull/32208

Sgitario commented:
PR updated by adding the ImageStream and BuildConfig resources.

gsmet commented:
@iocanel I let you approve and merge if you're happy with the changes.

iocanel reviewed (approved):
lgtm
